.github/scripts/parse-unit-test-output.py

```diff
@@ -0,0 +1,119 @@
+"""
+This script parses the Ant unit test output and identifies tests which have
+failed. It then retrieves their error messages from the surefire runner log
+directory. It also prints out the longest-running unit tests, for possible
+action by maintainers. Finally, it detects whether any tests ran multiple
+times by accident.
+
+The first argument should be a path to a file containing the captured output
+of the Ant unit tests. The second argument should be a path to the surefire
+runner log directory.
+"""
```

Comment:
Is there a way to detect failed tests without this python script?  Let's not make python a dependency of a local development environment.

Reply:
Yes, I was worried about that - the whole `fail` target thing ensures that python is not involved in whether a nonzero exit code is returned to the CI.